### PR TITLE
[REVIEW] Add support for `dummy_na` in `get_dummies`

### LINKED ISSUE #5542: [FEA] `get_dummies()` method has a Non Functional argument, `dummy_na`
**Is your feature request related to a problem? Please describe.**

`get_dummies()` method has a Non Functional argument, `dummy_na`

**Describe the solution you'd like**

Add support for the argument

**Describe alternatives you've considered**

Our own implementation for cuDF:
```python
import cudf as pd
def get_dummies(dataset, **kwargs):
    encoded = pd.get_dummies(dataset, **kwargs)
    encoded = encoded[[column for column in encoded.columns if "None" not in column]]
    return encoded
```

used with cuDF:
```python
df = pd.DataFrame({"a": ["value1", "value2", None], "b": [0, 0, 0]})
get_dummies(df)
```

**Additional context**

This is how it works in Pandas: 
```python
import pandas as pd
df = pd.DataFrame({"a": ["value1", "value2", None], "b": [0, 0, 0]})
pd.get_dummies(df)

```

### PR DESCRIPTION
**Pull request by rgsl888prabhu**
closes #5542

### CODE REVIEW

**Comment by GPUtester** [comment]
Please update the changelog in order to start CI tests.

View the gpuCI docs [here](https://docs.rapids.ai/gpuci#open-pr-or-update-pr).

**Comment by GPUtester** [comment]
Please update the changelog in order to start CI tests.

View the gpuCI docs [here](https://docs.rapids.ai/gpuci#open-pr-or-update-pr).

File: python/cudf/cudf/core/reshape.py [hunk 1]
```diff
@@ -449,11 +467,16 @@ def length_check(obj, name):
...
         result_df = df.drop(labels=columns)
         for name in columns:
-            if hasattr(df[name]._column, "categories"):
+            if isinstance(df[name]._column, CategoricalColumn):
                 unique = df[name]._column.categories
             else:
                 unique = df[name].unique()
 
+            if dummy_na is False:
+                if not isinstance(unique, StringColumn):
```
Comment (shwina) [inline python/cudf/cudf/core/reshape.py:476]: Won't line 473 return a `Series`, so should we use `is_string_dtype()` instead here?

Comment (rgsl888prabhu) [reply python/cudf/cudf/core/reshape.py:476]: Nice catch

File: python/cudf/cudf/core/reshape.py [hunk 2]
```diff
@@ -394,11 +402,24 @@ def get_dummies(
...
+    >>> import cudf
+    >>> df = cudf.DataFrame({"a": ["value1", "value2", None], "b": [0, 0, 0]})
+    >>> cudf.get_dummies(df)
+       b  a_value1  a_value2
+    0  0         1         0
+    1  0         0         1
+    2  0         0         0
+
+    >>> cudf.get_dummies(df, dummy_na=True)
+       b  a_None  a_value1  a_value2
```
Comment (kkraus14) [inline python/cudf/cudf/core/reshape.py:418]: Is `a_None` the naming scheme we want to use? Pandas uses `a_nan` it looks like for example. Given we can have both `NaN` and `null` values, I'd propose we use `a_nan` for `NaN` values and `a_null` for null values.

File: python/cudf/cudf/core/reshape.py [hunk 3]
```diff
@@ -449,11 +470,16 @@ def length_check(obj, name):
     else:
         result_df = df.drop(labels=columns)
         for name in columns:
-            if hasattr(df[name]._column, "categories"):
+            if isinstance(df[name]._column, CategoricalColumn):
                 unique = df[name]._column.categories
             else:
                 unique = df[name].unique()
 
+            if dummy_na is False:
```
Comment (kkraus14) [inline python/cudf/cudf/core/reshape.py:478]: `0 is False` evaluates to `False`:
```suggestion
            if not dummy_na:
```

File: python/cudf/cudf/core/reshape.py [hunk 4]
```diff
@@ -449,11 +470,16 @@ def length_check(obj, name):
...
+                if not is_string_dtype(unique):
+                    unique = unique.nans_to_nulls()
```
Comment (kkraus14) [inline python/cudf/cudf/core/reshape.py:480]: Shouldn't we only need to do this if `unique` is a `float` type?

Comment (rgsl888prabhu) [reply python/cudf/cudf/core/reshape.py:480]: I was thinking about categorical as well, but as we have already filtered that out, only float should suffice. Nice, catch.

**Pull request merged**